backported loading dl extension with -dexetnsion=extension.so from php5.1-200512091130

--- php-5.0.5/sapi/cli/php_cli.c	2005-03-22 17:09:20.000000000 +0200
+++ php5.1-200512091130/sapi/cli/php_cli.c	2005-12-06 03:38:26.000000000 +0200
@@ -409,7 +433,14 @@
 	} else {
 		value = "1";
 	}
-	zend_alter_ini_entry(name, strlen(name)+1, value, strlen(value), PHP_INI_SYSTEM, PHP_INI_STAGE_ACTIVATE);
+
+	if (!strcasecmp(name, "extension")) { /* load function module */
+		zval extension, zval;
+		ZVAL_STRING(&extension, value, 0);
+		php_dl(&extension, MODULE_PERSISTENT, &zval TSRMLS_CC);
+	} else {
+		zend_alter_ini_entry(name, strlen(name)+1, value, strlen(value), PHP_INI_SYSTEM, PHP_INI_STAGE_ACTIVATE);
+	}
 }
 
 
